File: rust/ds_algo/src/time_complexity.rs
use chrono::Utc;

pub fn time_complexity_example() {
    let num: i64 = get_current_timestamp();
    
    let start_time = get_current_timestamp();
    let sum_by_formula = find_sum_by_formula(num);
    let end_time = get_current_timestamp();
    println!(" Find sum of {} number by formula: {}. Took {} milliseconds", num, sum_by_formula, end_time - start_time);

    let start_time = get_current_timestamp();
    let sum_by_iteration = find_sum_by_iteration(num);
    let end_time = get_current_timestamp();
    println!(" Find sum of {} number by iteration: {}. Took {} milliseconds", num, sum_by_iteration, end_time - start_time);
}

fn find_sum_by_formula(n: i64) -> i64 {
    return n * (n + 1) / 2;
}

fn find_sum_by_iteration(n: i64) -> i64 {
    let mut sum: i64 = 0;

    for num in 1..=n {
        sum += num;
    }

    return sum;
}

fn get_current_timestamp() -> i64 {
    let current_time = Utc::now();
    let timestamp = current_time.timestamp();
    return timestamp;
}

File: kotlin/ds_algo/src/main/kotlin/examples/TimeComplexity.kt
package examples

import java.time.Instant

object TimeComplexity {
    fun timeComplexityExample() {
        val num: Long = getCurrentTimestamp()

        val startTime = getCurrentTimestamp()
        findSumByFormula(num)
        val endTime = getCurrentTimestamp()
        println(" Find sum of $num number by formula. Took ${endTime - startTime} milliseconds")

        val startTime2 = getCurrentTimestamp()
        findSumByIteration(num)
        val endTime2 = getCurrentTimestamp()
        println(" Find sum of $num number by iteration. Took ${endTime2 - startTime2} milliseconds")
    }

    private fun findSumByFormula(n: Long): Long {
        return n * (n + 1) / 2
    }

    private fun findSumByIteration(n: Long): Long {
        var sum: Long = 0
        for (num in 1..n) {
            sum += num
        }
        return sum
    }

    private fun getCurrentTimestamp(): Long {
        return Instant.now().epochSecond
    }
}